lib/std/os/windows.zig

```diff
@@ -558,6 +558,46 @@ pub fn WriteFile(
     }
 }
 
+pub const SetCurrentDirectoryError = error{
+    NameTooLong,
+    InvalidUtf8,
+    FileNotFound,
+    NotDir,
+    AccessDenied,
+    NoDevice,
+    Unexpected,
+};
+
+pub fn SetCurrentDirectory(path_name: []const u8) SetCurrentDirectoryError!void {
+    var path_space: PathSpace = undefined;
+    path_space.len = try std.unicode.utf8ToUtf16Le(path_space.data[0..], path_name);
```

Comment:
If we are to be consistent, this function should take a buffer of UTF16 instead of UTF8, and leave the conversion to be performed one level up. This will also clean up error values in the sense that `InvalidUtf8` will not be required at this level at all.

This conversion also doesn't make exhaustive checks and path conversions required by NT. In fact, you should really use `windows.sliceToPrefixedFileW` which performs the conversion from `[]const u8` to `windows.PathSpace` for you.

Reply:
My first attempt used `windows.sliceToPrefixedFileW` however when using that completely valid paths return a  `NTSTATUS.OBJECT_NAME_INVALID`.

However without doing anything other than pure utf8 to utf16 (as in the pull request) the exact same paths correctly change the cwd.

Reply:
Unless im doing something wrong? This is the first time i've interacted with the NtDll api's.
The code below will give `NTSTATUS.OBJECT_NAME_INVALID` for a valid path that will work fine with the code in the pr.
(I know the utf8 -> utf16 should be one level up this is just as an example)

```zig
pub fn SetCurrentDirectory(path_name: []const u8) SetCurrentDirectoryError!void {
    const dir_path_w = try sliceToPrefixedFileW(path_name);
        
    const path_len_bytes = math.cast(u16, dir_path_w.len * 2) catch |err| switch (err) {
        error.Overflow => return error.NameTooLong,
    };
    
    var nt_name = UNICODE_STRING{
        .Length = path_len_bytes,
        .MaximumLength = path_len_bytes,
        .Buffer = @intToPtr([*]u16, @ptrToInt(&dir_path_w.data)),
    };
    
    const rc = ntdll.RtlSetCurrentDirectory_U(&nt_name);
    switch (rc) {
        .SUCCESS => {},
        .OBJECT_NAME_INVALID => unreachable,
        .OBJECT_NAME_NOT_FOUND => return error.FileNotFound,
        .OBJECT_PATH_NOT_FOUND => return error.FileNotFound,
        .NO_MEDIA_IN_DEVICE => return error.NoDevice,
        .INVALID_PARAMETER => unreachable,
        .ACCESS_DENIED => return error.AccessDenied,
        .OBJECT_PATH_SYNTAX_BAD => unreachable,
        .NOT_A_DIRECTORY => return error.NotDir,
        else => return unexpectedStatus(rc),
    }
}
```